server/src/main/java/org/candlepin/resource/CdnResource.java

```diff
@@ -88,36 +96,110 @@ public void delete(@PathParam("label") String label,
     @POST
     @Consumes(MediaType.APPLICATION_JSON)
     @Produces(MediaType.APPLICATION_JSON)
-    public Cdn create(
-        @ApiParam(name = "cdn", required = true) Cdn cdn,
+    public CdnDTO create(
+        @ApiParam(name = "cdn", required = true) CdnDTO cdnDTOInput,
         @Context Principal principal) {
-        Cdn existing = curator.lookupByLabel(cdn.getLabel());
+        Cdn existing = curator.lookupByLabel(cdnDTOInput.getLabel());
         if (existing != null) {
-            throw new BadRequestException(i18n.tr("A CDN with the label {0} already exists", cdn.getLabel()));
+            throw new BadRequestException(i18n.tr(
+                "A CDN with the label {0} already exists", cdnDTOInput.getLabel()));
         }
-        return cdnManager.createCdn(cdn);
+
+        Cdn cndToCreate = new Cdn();
+        this.populateEntity(cndToCreate, cdnDTOInput);
+
+        return this.translator.translate(cdnManager.createCdn(cndToCreate), CdnDTO.class);
     }
 
     @ApiOperation(notes = "Updates a CDN @return a Cdn object", value = "update")
     @PUT
     @Consumes(MediaType.APPLICATION_JSON)
     @Produces(MediaType.APPLICATION_JSON)
     @Path("/{label}")
-    public Cdn update(@PathParam("label") String label,
-        @ApiParam(name = "cdn", required = true) Cdn cdn,
+    public CdnDTO update(@PathParam("label") String label,
+        @ApiParam(name = "cdn", required = true) Cdn cdnDTOInput,
```

Comment:
Missed this one the first time around: we're still accepting an entity on input -- this should accept a CdnDTO, which will likely mean the rest of this method changes as well.

Also, given that we have a CdnManager already, this is probably a good time to update this resource to only do validation and offload any actual work to the manager.